llvm/unittests/tools/llvm-exegesis/RISCV/TargetTest.cpp

```diff
@@ -0,0 +1,62 @@
+//===-- TargetTest.cpp ---------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "Target.h"
+
+#include <cassert>
+#include <memory>
+
+#include "MCTargetDesc/RISCVMCTargetDesc.h"
+#include "llvm/MC/TargetRegistry.h"
+#include "llvm/Support/TargetSelect.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace llvm{
+namespace exegesis {
+
+void InitializeRISCVExegesisTarget();
+
+namespace {
+
+using testing::NotNull;
+using testing::IsEmpty;
+using testing::Not;
+
+constexpr const char kTriple[] = "riscv64-unknown-linux";
+
+class RISCVTargetTest : public ::testing::Test {
```

Comment:
I'm wondering if this should inherit from `RISCVTestBase`? Looking at all the other targets, it doesn't seem like we do that anywhere, but having the general setup in a centralized spot might be nice...

Reply:
I think Mips does have it inherit from MipsTestBase. My patch was a bit of a weird combination of PowerPC and Mips. My original goal was to get the Pfm counter in, then I discovered we had less unittests than other targets.

Reply:
I think we should probably prefer inheriting from the test base class. I'll send a patch to make the others consistent.